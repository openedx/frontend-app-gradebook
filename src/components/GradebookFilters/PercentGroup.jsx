/* eslint-disable react/sort-comp */
import React from 'react';
import PropTypes from 'prop-types';

import { Form } from '@edx/paragon';

const PercentGroup = ({
  id,
  label,
  value,
  disabled,
  onChange,
}) => (
  <div className="percent-group">
    <Form.Group controlId={id}>
      <Form.Label>{label}</Form.Label>
      <Form.Control
        type="number"
        min={0}
        max={100}
        step={1}
        {...{ value, disabled, onChange }}
      />
    </Form.Group>
    <span className="input-percent-label">%</span>
  </div>
);
PercentGroup.defaultProps = {
  disabled: false,
};
PercentGroup.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.node.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
};

export default PercentGroup;
