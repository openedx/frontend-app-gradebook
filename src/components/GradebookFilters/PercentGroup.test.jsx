import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';

import PercentGroup from './PercentGroup';

describe('PercentGroup', () => {
  let props = {
    id: 'group id',
    label: 'Group Label',
    value: 'group VALUE',
    disabled: false,
  };

  beforeEach(() => {
    props = {
      ...props,
      onChange: jest.fn().mockName('props.onChange'),
    };
  });

  describe('Component', () => {
    describe('snapshots', () => {
      test('basic snapshot', () => {
        const el = shallow(<PercentGroup {...props} />);
        expect(el.snapshot).toMatchSnapshot();
      });
      test('disabled', () => {
        const el = shallow(<PercentGroup {...props} disabled />);
        expect(el.snapshot).toMatchSnapshot();
      });
    });
  });
});
