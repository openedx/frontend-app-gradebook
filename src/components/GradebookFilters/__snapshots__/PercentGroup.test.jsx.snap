// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PercentGroup Component snapshots basic snapshot 1`] = `
<div
  className="percent-group"
>
  <Form.Group
    controlId="group id"
  >
    <Form.Label>
      Group Label
    </Form.Label>
    <Form.Control
      disabled={false}
      max={100}
      min={0}
      onChange={[MockFunction props.onChange]}
      step={1}
      type="number"
      value="group VALUE"
    />
  </Form.Group>
  <span
    className="input-percent-label"
  >
    %
  </span>
</div>
`;

exports[`PercentGroup Component snapshots disabled 1`] = `
<div
  className="percent-group"
>
  <Form.Group
    controlId="group id"
  >
    <Form.Label>
      Group Label
    </Form.Label>
    <Form.Control
      disabled={true}
      max={100}
      min={0}
      onChange={[MockFunction props.onChange]}
      step={1}
      type="number"
      value="group VALUE"
    />
  </Form.Group>
  <span
    className="input-percent-label"
  >
    %
  </span>
</div>
`;
